src/cli/keepassxc-cli.cpp

```diff
@@ -0,0 +1,108 @@
+ */
+
+#include <cstdlib>
+
+#include <QCommandLineParser>
+#include <QCoreApplication>
+#include <QStringList>
+
+#include <cli/EntropyMeter.h>
+#include <cli/Extract.h>
+#include <cli/List.h>
+#include <cli/Merge.h>
+#include <cli/Show.h>
+
+#include "config-keepassx.h"
+#include "core/Tools.h"
+#include "crypto/Crypto.h"
+
+int main(int argc, char **argv)
+{
+#ifdef QT_NO_DEBUG
+    Tools::disableCoreDumps();
+#endif
+
+    if (!Crypto::init()) {
+        qFatal("Fatal error while testing the cryptographic functions:\n%s", qPrintable(Crypto::errorString()));
+        return EXIT_FAILURE;
+    }
+
+    QCoreApplication app(argc, argv);
+    app.setApplicationVersion(KEEPASSX_VERSION);
+
+    QCommandLineParser parser;
+
+    QString description("KeePassXC command line interface.");
+    description = description.append(QString("\n\nAvailable commands:"));
+    description = description.append(QString("\n  extract\tExtract and print the content of a database."));
+    description = description.append(QString("\n  entropy-meter\tCalculate password entropy."));
+    description = description.append(QString("\n  list\t\tList database entries."));
+    description = description.append(QString("\n  merge\t\tMerge two databases."));
+    description = description.append(QString("\n  show\t\tShow a password."));
+    parser.setApplicationDescription(QCoreApplication::translate("main", qPrintable(description)));
+
+    parser.addPositionalArgument("command", QCoreApplication::translate("main", "Name of the command to execute."));
+
+    parser.addHelpOption();
+    parser.addVersionOption();
+    // TODO : use process once the setOptionsAfterPositionalArgumentsMode (Qt 5.6)
+    // is available. Until then, options passed to sub-commands won't be
+    // recognized by this parser.
+    // parser.process(app);
+
+    if (argc < 2) {
+        parser.showHelp();
+        return EXIT_FAILURE;
+    }
+
+    QString commandName = argv[1];
+
+    // Removing the first cli argument before dispatching.
+    ++argv;
+    --argc;
+
+    if (commandName == "entropy-meter") {
+        argv[0] = const_cast<char*>("keepassxc-cli entropy-meter");
+        return EntropyMeter::execute(argc, argv);
+    }
+
+    if (commandName == "extract") {
+        argv[0] = const_cast<char*>("keepassxc-cli extract");
+        return Extract::execute(argc, argv);
+    }
+
+    if (commandName == "list") {
+        argv[0] = const_cast<char*>("keepassxc-cli list");
+        return List::execute(argc, argv);
+    }
+
+    if (commandName == "merge") {
+        argv[0] = const_cast<char*>("keepassxc-cli merge");
+        return Merge::execute(argc, argv);
+    }
+
+    if (commandName == "show") {
+        argv[0] = const_cast<char*>("keepassxc-cli show");
+        return Show::execute(argc, argv);
+    }
+
+    qCritical("Invalid command %s.", qPrintable(commandName));
+    parser.showHelp();
+    return EXIT_FAILURE;
```

Comment:
You should add the lsan stuff I added to src/main.cpp to avoid a  huge dump of memory leak errors when compiling with our new -DWITH_ASAN=ON option.